Failed tests (11198): TestFailedReplicaChange

The following test appears to have failed:

[#11198](https://circleci.com/gh/cockroachdb/cockroach/11198):

```
I0113 08:41:42.068364 951 kv/range_cache.go:128  lookup range descriptor: key=/Meta2/"\x93\x88"
I0113 08:41:42.069046 951 kv/range_cache.go:189  adding descriptor: key=/Meta2/Max desc=range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I0113 08:41:42.069123 951 gossip/gossip.go:218  setting node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<> 
I0113 08:41:42.070880 951 kv/range_cache.go:189  adding descriptor: key=/Meta2/Max desc=range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
==================
WARNING: DATA RACE
Write by goroutine 128:
  github.com/cockroachdb/cockroach/storage_test.TestFailedReplicaChange()
      /go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:375 +0x209
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:456 +0xdc

Previous read by goroutine 113:
  github.com/cockroachdb/cockroach/storage.(*Replica).executeCmd()
      /go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:58 +0x2ab
  github.com/cockroachdb/cockroach/storage.(*Replica).executeBatch()
      /go/src/github.com/cockroachdb/cockroach/storage/replica.go:1497 +0x38a
I0113 08:41:42.071183 951 kv/range_cache.go:313  clearing overlapping descriptor: key=/Meta2/Max desc=range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
  github.com/cockroachdb/cockroach/storage.(*Replica).addReadOnlyCmd()
      /go/src/github.com/cockroachdb/cockroach/storage/replica.go:850 +0x262
  github.com/cockroachdb/cockroach/storage.(*Replica).Send()
      /go/src/github.com/cockroachdb/cockroach/storage/replica.go:660 +0x540
  github.com/cockroachdb/cockroach/storage.(*Store).Send()
--
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:333 +0x21d
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x5f

Goroutine 128 (running) created at:
  testing.RunTests()
      /tmp/workdir/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /tmp/workdir/go/src/testing/testing.go:494 +0xe4
  github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck()
      /go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x2e
  github.com/cockroachdb/cockroach/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x33
  main.main()
      github.com/cockroachdb/cockroach/storage/_test/_testmain.go:504 +0x209

Goroutine 113 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x6f
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:349 +0x105
  github.com/cockroachdb/cockroach/storage.(*baseQueue).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:202 +0x42
  github.com/cockroachdb/cockroach/storage.(*splitQueue).Start()
      <autogenerated>:74 +0x5f
  github.com/cockroachdb/cockroach/storage.(*replicaScanner).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:106 +0xd3
--
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:588 +0x160
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x5f
==================
==================
WARNING: DATA RACE
Read by goroutine 113:
  github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).rpcSend()
      /go/src/github.com/cockroachdb/cockroach/storage/client_test.go:321 +0x3cd
  github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).(github.com/cockroachdb/cockroach/storage_test.rpcSend)-fm()
      /go/src/github.com/cockroachdb/cockroach/storage/client_test.go:228 +0x115
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:372 +0x7b7
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:450 +0x275
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendChunk.func1()
--
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:333 +0x21d
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x5f

Previous write by goroutine 128:
  github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/storage/client_test.go:434 +0x16ed
  github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/client_test.go:235 +0x17a1
  github.com/cockroachdb/cockroach/storage_test.startMultiTestContext()
      /go/src/github.com/cockroachdb/cockroach/storage/client_test.go:190 +0x8b
  github.com/cockroachdb/cockroach/storage_test.TestFailedReplicaChange()
      /go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:369 +0xe1
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:456 +0xdc

Goroutine 113 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x6f
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:349 +0x105
  github.com/cockroachdb/cockroach/storage.(*baseQueue).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/queue.go:202 +0x42
  github.com/cockroachdb/cockroach/storage.(*splitQueue).Start()
      <autogenerated>:74 +0x5f
  github.com/cockroachdb/cockroach/storage.(*replicaScanner).Start()
      /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:106 +0xd3
  github.com/cockroachdb/cockroach/storage.(*Store).Start.func4()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:588 +0x160
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x5f

Goroutine 128 (running) created at:
  testing.RunTests()
      /tmp/workdir/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /tmp/workdir/go/src/testing/testing.go:494 +0xe4
  github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck()
      /go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x2e
  github.com/cockroachdb/cockroach/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x33
  main.main()
      github.com/cockroachdb/cockroach/storage/_test/_testmain.go:504 +0x209
--
I0113 08:41:42.101281 951 storage/replica.go:1031  raft: proposing ADD_REPLICA {2 2 2} for range 1
W0113 08:41:42.104370 951 kv/dist_sender.go:635  failed to invoke BeginTransaction [/Local/Range/RangeDescriptor/"",/Min), ConditionalPut [/Local/Range/RangeDescriptor/"",/Min), Put [/Meta2/Max,/Min), Put [/Meta1/Max,/Min), EndTransaction [/Local/Range/RangeDescriptor/"",/Min): unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:35 > 
I0113 08:41:42.105109 951 client/db.go:485  failed batch: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:35 > 
I0113 08:41:42.107071 951 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0113 08:41:42.107269 951 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestFailedReplicaChange (0.06s)
	client_raft_test.go:419: storage/replica_command.go:1737: change replicas of 1 failed: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:35 > 
=== RUN   TestReplicateAfterTruncation
E0113 08:41:42.110914 951 gossip/gossip.go:210  different node IDs were set for the same gossip instance (2147483647, 1)
I0113 08:41:42.111372 951 raft/raft.go:446  [group 1] 1 became follower at term 5
I0113 08:41:42.114717 951 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0113 08:41:42.114848 951 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0113 08:41:42.114965 951 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0113 08:41:42.115084 951 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0113 08:41:42.115216 951 raft/raft.go:487  [group 1] 1 became leader at term 6
I0113 08:41:42.116949 951 storage/replica.go:1611  gossiping cluster id  from store 1, range 1
--
I0113 08:42:01.074413 951 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestLogSplits (1.03s)
	<autogenerated>:28: prior leak detected, leaktest disabled
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.72s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	25.550s
=== RUN   TestBatchBasics
I0113 08:42:03.340261 1016 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0113 08:42:03.344203 1016 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I0113 08:42:03.350896 1016 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -366,9 +366,6 @@ func TestFailedReplicaChange(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	defer func() { storage.TestingCommandFilter = nil }()
 
-	mtc := startMultiTestContext(t, 2)
-	defer mtc.Stop()
-
 	var runFilter atomic.Value
 	runFilter.Store(true)
 
@@ -381,6 +378,9 @@ func TestFailedReplicaChange(t *testing.T) {
 		}
 		return nil
 	}
+
+	mtc := startMultiTestContext(t, 2)
+	defer mtc.Stop()
 
 	rng, err := mtc.stores[0].GetReplica(1)
 	if err != nil {

